dashboard/app/models/user.rb

```diff
@@ -895,7 +916,10 @@ def self.find_for_authentication(tainted_conditions)
       ).first
     elsif hashed_email = conditions.delete(:hashed_email)
       return nil if hashed_email.utf8mb4?
-      where(hashed_email: hashed_email).first
+      legacy = where(hashed_email: hashed_email).first
+      matching_auth_opts = AuthenticationOption.where(hashed_email: hashed_email)
+      migrated = matching_auth_opts.first && matching_auth_opts.first.user
+      return migrated || legacy
```

Comment:
This logic shows up twice in this diff - once in `find_by_email_or_hashed_email` and once here.  Should we extract a `find_by_hashed_email` helper that we use in both places?  _And_ can we change this logic to entirely avoid the query for  the legacy user if a migrated user is found?

Reply:
:+1: I'm merging staging in so I pick that change up, and I'll use it.